auth0_client.js

```diff
@@ -51,45 +81,200 @@ Auth0.requestCredential = function(options, credentialRequestCompleteCallback) {
     clientId: Meteor.settings.public.AUTH0_CLIENT_ID,
     hostname: Meteor.settings.public.AUTH0_DOMAIN,
     loginStyle: 'redirect',
-  };
+  }
 
   /**
    * Boilerplate
    */
-  const credentialToken = Random.secret();
-  const loginStyle = OAuth._loginStyle('auth0', config, options);
-  let path = options.path || '';
-  path = path.startsWith('/') ? path.substring(1) : path;
-  
+
+  // Create one-time credential secret token
+  const credentialToken = Random.secret()
+
+  // Detemines the login style
+  const loginStyle = Auth0._loginStyle(config, options)
+  const rootUrl = Auth0._rootUrl(options)
+  const redirectUrl = `${rootUrl}_oauth/auth0`
+
+  // Determine path
+  let path = options.path || ''
+  path = path.startsWith('/') ? path.substring(1) : path
+  const callbackUrl = `${rootUrl}${path}`
+
   /**
    * Imgur requires response_type and client_id
    * We use state to roundtrip a random token to help protect against CSRF (boilerplate)
    */
-  let loginUrl = `https://${config.hostname}/authorize/` +
+
+  let loginUrl =
+    `https://${config.hostname}/authorize/` +
     '?scope=openid%20profile%20email' +
     '&response_type=code' +
-    '&client_id=' + config.clientId +
-    '&state=' + OAuth._stateParam(loginStyle, credentialToken, `${Meteor.absoluteUrl('')}${path}`) +
-    // '&connection=facebook' +
-    
-    `&redirect_uri=${Meteor.absoluteUrl('_oauth/auth0')}`
-    ;
+    '&client_id=' +
+    config.clientId +
+    '&state=' +
+    OAuth._stateParam(
+      loginStyle === 'inline' ? 'redirect' : loginStyle,
```

Comment:
just use 'redirect' as flag instead of 'inline', so you can just give back loginStyle i think.